frontend/js/main.js

```diff
@@ -0,0 +1,2203 @@
+    }
+    tableBody.appendChild(tableRow);
+  }
+  globalDataTypeTable.appendChild(tableBody);
+  globalDataTypeDiv = document.getElementById('globalDataType');
+  globalDataTypeDiv.innerHTML = '';
+  globalDataTypeDiv.appendChild(globalDataTypeTable);
+  tooltipHandler();
+}
+
+/**
+ * Function to update data types with warnings(if any) in global data type table
+ *
+ * @param {string} id id of select box in global data type table
+ * @return {null}
+ */
+const dataTypeUpdate = (id) => {
+  idNum = parseInt(id.match(/\d+/), 10);
+  dataTypeOptionArray = globalDataTypeList[document.getElementById('dataTypeKey' + idNum).innerHTML];
+
+  optionHTML = '';
+  selectHTML = `<div style='display: flex;'>
+                  <i class="large material-icons warning" style='cursor: pointer; visibility: hidden;'>warning</i>
+                  <select onchange='dataTypeUpdate(id)' class='form-control tableSelect' id=${id} style='border: 0px !important;'>
+                </div>`;
+  for (var x = 0; x < dataTypeOptionArray.length; x++) {
+    optionFound = false;
+    if (dataTypeOptionArray[x].T == document.getElementById(id).value) {
+      optionFound = true;
+    }
+    if (dataTypeOptionArray[x].T == document.getElementById(id).value && dataTypeOptionArray[x].Brief != "") {
+
+      selectHTML = `<div style='display: flex;'>
+                      <i class="large material-icons warning" style='cursor: pointer;' data-toggle='tooltip' data-placement='bottom' title='${dataTypeOptionArray[x].Brief}'>warning</i>
+                      <select onchange='dataTypeUpdate(id)' class='form-control tableSelect' id=${id} style='border: 0px !important;'>
+                    </div>`;
+    }
+    if (optionFound == true) {
+      optionHTML += `<option selected='selected' value='${dataTypeOptionArray[x].T}'>${dataTypeOptionArray[x].T} </option>`;
+    }
+    else {
+      optionHTML += `<option value='${dataTypeOptionArray[x].T}'>${dataTypeOptionArray[x].T} </option>`;
+    }
+  }
+  selectHTML += optionHTML + `</select>`;
+  document.getElementById('dataTypeVal' + idNum).innerHTML = selectHTML;
+  tooltipHandler();
+}
+
+/**
+ * Function to create table from con json structure
+ *
+ * @param {json} obj Json object contaning source and spanner table information
+ * @return {null}
+ */
+async function createSourceAndSpannerTables(obj) {
+  getFilePaths();
+  schemaConversionObj = obj;
+
+  // fetch global data type list
+  fetch(apiUrl + '/typemap', {
+    method: 'GET',
+    headers: {
+      'Accept': 'application/json',
+      'Content-Type': 'application/json'
+    }
+  })
+  .then(function (res) {
+    console.log(res.ok);
+    if (res.ok) {
+      res.json().then(function (result) {
+        globalDataTypes = result;
+        localStorage.setItem('globalDataTypeList', JSON.stringify(globalDataTypes));
+      });
+    }
+    else {
+      return Promise.reject(res);
+    }
+  }).catch(function (err) {
+    showSnackbar(err, ' redBg');
+  });
```

Comment:
@13shreya I think it's good to create a common fetch function for api calls and pass just apiUrl.

Better to create a separate file called api.js and place all the server calls in that file. Let's do that and let me know if you need any help.